src/js/experiments/snippetFeedback.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * The SnippetFeedback class is responsible for dynamically adding a user
+ * feedback form whenever a user clicks on the "copy code" button.
+ *
+ * This class is instantiated for each snippet in a page.
+ * Most of the methods are for dynamically adding dom elements for
+ * capturing user feedback.
+ *
+ * */
+export class SnippetFeedback {
+  constructor(snippetElement) {
+    // Dom elements
+    this.snippetElement = snippetElement;
+    this.codeSnippetContainer = null;
+    this.wasThisHelpfulContainer = null;
+    this.feedbackForm = null;
+    // For tracking user's response to be included in the final feedback form.
+    this.wasThisHelpful = null;
+    this.renderWasThisHelpfulPrompt();
+  }
+
+  // -- Render functions
+  // These are responsible for constructing and attaching new dom elements
+  // based on the user's flow in the feedback process.
+  //
+
+  /**
+   * Construct a "Was this helpful" line, and inject it below the code snippet.
+   */
+  renderWasThisHelpfulPrompt() {
+    this.codeSnippetContainer = this.snippetElement.closest('div.highlight');
+    // Create some dom elements:
+    const container = this.makeElement({
+      kind: 'div',
+      className: 'was-this-helpful',
+    });
+    const textPrompt = this.makeElement({
+      kind: 'span',
+      text: 'Was this helpful?',
+    });
+    const slash = this.makeElement({ kind: 'span', text: ' / ' });
+    const yesButton = this.renderYesNoButton({ text: 'Yes' });
+    const noButton = this.renderYesNoButton({ text: 'No' });
+    // append dom nodes.
+    container.appendChild(textPrompt);
+    container.appendChild(yesButton);
+    container.appendChild(slash);
+    container.appendChild(noButton);
+    this.wasThisHelpfulContainer = container;
+    this.codeSnippetContainer.appendChild(container);
+  }
+
+  /**
+   * Creates the dom element for the yes/no button in the prompt.
+   * */
+  renderYesNoButton({ text }) {
+    let yesBtn = this.makeElement({
+      kind: 'button',
+      text,
+      className: 'text-btn',
+      onClick: () => {
+        this.wasThisHelpful = text;
+        this.trackYesOrNoButton(text);
+        // only pop up the form if it has not yet been created.
+        if (this.feedbackForm == null) {
+          this.renderFeedbackForm();
+        }
+      },
+    });
+
+    return yesBtn;
+  }
+
+  /**
+   * Renders a feedback form to the user, provided they have clicked
+   * on a "yes" or "no" button in the 'was this helpful' prompt.
+   * */
+  renderFeedbackForm() {
+    if (!this.feedbackForm) {
+      const container = this.makeElement({ kind: 'div', className: 'form' });
+      const prompt = this.makeElement({
+        kind: 'div',
+        text: 'Any way we can improve?',
```

Comment:
If its not too much effort, could we include a different message between `yes` and `no`? I feel like having the same message for both could be a bit confusing. Maybe something like `'How can we improve'` if users click `no`.

Reply:
I'm going to defer to design on that one, but I think this is a good point.

Reply:
<img width="1085" alt="Screen Shot 2021-12-01 at 2 01 43 PM" src="https://user-images.githubusercontent.com/57234605/144305293-7604a69d-98ed-4141-bd31-9e44ab743fa9.png">